### PR TITLE
feat: added and updated index export tracking (#222)

### PR DESCRIPTION
**Pull request by jpaten**
- Added event models as follows:
- Add the following event models
  - `index_analyze_in_terra_requested`
    - `entity_name`
  - `index_file_manifest_requested`
    - `entity_name`
 - Alter the existing `bulk_download_requested` model as follows
   -  Remove params for `catalog` and `current_query`
   -  Rename `index` to `entity_name`
 - Called events on clicking the request button
 - Documented changes in `readme-analytics.md`

### CODE REVIEW

**Comment by NoopDog** [comment]
Hi @jpaten is this still meant to be in WIP or is it ready to review? Thx!

**Comment by MillenniumFalconMechanic** [comment]
Hi @jpaten, naming conventions here LGTM. Thank you! cc @NoopDog.

File: src/components/Export/common/tracking.ts [hunk 1]
```diff
@@ -3,27 +3,32 @@ import { EVENT_NAME, EVENT_PARAM } from "../../../common/analytics/entities";
...
-    [EVENT_PARAM.INDEX]: index,
+    [EVENT_PARAM.ENTITY_NAME]: entity_name,
     [EVENT_PARAM.TOOL_NAME]: toolName,
   });
 }
+
+export function fileManifestTracking(entity_name: string): void {
+  // Track the file manifest requested event.
+  console.log("tracking file manifest requested event");
+  console.log(entity_name);
```
Comment (MillenniumFalconMechanic) [inline src/components/Export/common/tracking.ts:23]: Remove console.log's!

File: src/components/Export/common/tracking.ts [hunk 2]
```diff
@@ -3,27 +3,32 @@ import { EVENT_NAME, EVENT_PARAM } from "../../../common/analytics/entities";
...
   track(EVENT_NAME.BULK_DOWNLOAD_REQUESTED, {
-    [EVENT_PARAM.CATALOG]: catalog,
-    [EVENT_PARAM.CURRENT_QUERY]: currentQuery,
-    [EVENT_PARAM.INDEX]: index,
+    [EVENT_PARAM.ENTITY_NAME]: entity_name,
     [EVENT_PARAM.TOOL_NAME]: toolName,
   });
 }
+
+export function fileManifestTracking(entity_name: string): void {
```
Comment (MillenniumFalconMechanic) [inline src/components/Export/common/tracking.ts:20]: Can you add TS docs to `fileManifestTracking` and `exportToTerraTracking` similar to the `bulkDownloadTracking` function above?

**Review by MillenniumFalconMechanic — CHANGES_REQUESTED** [review]
Hi @jpaten, looks good! Just added a couple of nits.

File: src/components/Export/common/tracking.ts (hunk 1 above)
Comment (jpaten) [reply src/components/Export/common/tracking.ts:23]: Whoops! So sorry about that!

File: src/components/Export/common/tracking.ts (hunk 2 above)
Comment (jpaten) [reply src/components/Export/common/tracking.ts:20]: Will do!

**Comment by jpaten** [comment]
@MillenniumFalconMechanic sorry about those errors, I've updated the PR as suggested!

**Review by MillenniumFalconMechanic — APPROVED** [review]
LGTM, thanks Jonah!

**Pull request merged**